cpp/memilio/epidemiology/lct_populations.h

```diff
@@ -0,0 +1,212 @@
+class LctPopulations
+{
+public:
+    using Type                       = UncertainValue<FP>;
+    using InternalArrayType          = Eigen::Array<Type, Eigen::Dynamic, 1>;
+    using tupleLctStates             = std::tuple<LctStates...>;
+    static size_t constexpr m_groups = sizeof...(LctStates); ///< Number of groups.
+    static_assert(m_groups >= 1, "The number of LctStates provided should be at least one.");
+
+    /// @brief Default constructor.
+    LctPopulations()
+    {
+        set_count();
+        m_y = InternalArrayType::Constant(m_count, 1, 0.);
+    }
+
+    /**
+     * @brief get_num_compartments returns the number of compartments.
+     * @return Number of compartments which equals the flat array size.
+     */
+    size_t get_num_compartments() const
+    {
+        return m_count;
+    }
+
+    /**
+     * @brief Returns a reference to the internally stored flat array.
+     * @return Const reference to the InternalArrayType instance.
+     */
+    auto const& array() const
+    {
+        return m_y;
+    }
+    auto& array()
+    {
+        return m_y;
+    }
+
+    /**
+     * @brief Returns the entry of the array given a flat index index.
+     * @param index A flat index.
+     * @return The value of the internal array at the index.
+     */
+    Type& operator[](size_t index)
+    {
+        assert(index < m_count);
+        return m_y[index];
+    }
+
+    /**
+     @brief Gets the first index of a group in the flat array.
+    * @tparam group The group for which the index should be returned.
+    * @return The index of the first entry of group in the flat array.
+    */
+    template <size_t Group = 0, std::enable_if_t<(Group < m_groups) && (Group >= 0), bool> = true>
+    size_t get_first_index_group() const
+    {
+        if constexpr (Group == 0) {
+            return 0;
+        }
+        else {
+            return get_first_index_group<Group - 1>() + std::tuple_element_t<Group - 1, tupleLctStates>::Count;
+        }
+    }
+    /**
+     * @brief Returns an Eigen copy of the vector of populations. 
+     * This can be used as initial conditions for the ODE solver.
+     * @return Eigen::VectorXd of populations.
+     */
+    inline Vector<FP> get_compartments() const
+    {
+        return m_y.array().template cast<FP>();
+    }
+
+    /**
+     * @brief Returns the total population of a group.
+     * @tparam group The group for which the total population should be calculated.
+     * @return Total population of the group.
+     */
+    template <size_t group>
+    ScalarType get_group_total() const
+    {
+        return m_y.array()
+            .segment(get_first_index_group<group>(), std::tuple_element_t<group, tupleLctStates>::Count)
+            .sum();
+    }
+
+    /**
+     * @brief Returns the total population of all compartments and groups.
+     * @return Total population.
+     */
+    ScalarType get_total() const
+    {
+        return m_y.array().template cast<ScalarType>().sum();
```

Comment:
Why do you use ScalarType and not FP here?

Reply:
Actually, i dont know, just took it from the populations file. I think it is good to change this.